eth/handler.go

```diff
@@ -606,13 +606,30 @@ func (h *handler) BroadcastBlock(block *types.Block, propagate bool) {
 			log.Error("Propagating dangling block", "number", block.Number(), "hash", hash)
 			return
 		}
+
+		// These are the static and trusted peers which are not
+		// in `transfer := peers[:int(math.Sqrt(float64(len(peers))))]`
+		staticAndTrustedPeers := []*ethPeer{}
+
+		for _, peer := range peers[int(math.Sqrt(float64(len(peers)))):] {
+			if peer.IsTrusted() || peer.IsStatic() {
+				staticAndTrustedPeers = append(staticAndTrustedPeers, peer)
+			}
+		}
+
 		// Send the block to a subset of our peers
 		transfer := peers[:int(math.Sqrt(float64(len(peers))))]
 		for _, peer := range transfer {
 			peer.AsyncSendNewBlock(block, td)
 		}
 
-		log.Trace("Propagated block", "hash", hash, "recipients", len(transfer), "duration", common.PrettyDuration(time.Since(block.ReceivedAt)))
+		// Send the block to the trusted and static peers
+		for _, peer := range staticAndTrustedPeers {
+			log.Trace("Propagating block to static and trusted peer", "hash", hash, "peer", peer.ID())
```

Comment:
Shall we rename `peer` to `peerID`, for clarity?

Reply:
Actually `peer` here is the `*ethPeer` structure. `peerID` would be `peer.ID()`

Reply:
Yeah, I meant in the trace log, where you print `peer` as `peer.ID()`
Anyway, minor one

Reply:
Oops, my bad. Updated, thanks!